tutorials-v5/time-evolution/013_nonmarkovian_monte_carlo.md

```diff
@@ -0,0 +1,468 @@
+
+We consider two qubits that couple collectively to the same bosonic bath but do not interact with each other directly.
+The full model Hamiltonian is
+
+$$ H = \omega_1 \sigma^{(1)}_+\sigma^{(1)}_- + \omega_2 \sigma^{(2)}_+\sigma^{(2)}_- + \sum_k \epsilon_k b_k^\dagger b_k +\sum_k g_k [b_k(\sigma^{(1)}_++\sigma^{(2)}_+)+b_k^\dagger(\sigma^{(1)}_-+\sigma^{(2)}_-)]$$
+
+Here, $b_k$, $b_k^\dagger$ are bosonic ladder operators and $\sigma^{(j)}_\pm=(\sigma_x^{(j)}\pm i\sigma_y^{(j)})/2$ and $\sigma_{x,y,z}^{(j)}$ are the Pauli matrices acting on the $j$th qubit.
+Further, $\omega_j$ and $\epsilon_k$ denote the characteristic frequencies of the qubits and the bath modes, and $g_k$ are coupling constants.
+The initial state of the total system is $\rho_{2q}\otimes |0\rangle\langle 0|$, where $\rho_{2q}$ is the two-qubit initial state and $|0\rangle\langle 0|$ the boson bath vacuum state.
+
+Following Ref. [\[5\]](#References), the bath degrees can be integrated over such that the resulting two-qubit evolution is described by the Redfield master equation
+$$ \quad \frac{d}{dt}\rho(t) = -i \sum_{i,j=1}^2 A_{i,j} [\sigma_+^{(j)}\sigma_-^{(i)},\rho(t)] + \sum_{i,j=1}^2B_{i,j}\left( \sigma_-^{(i)}\rho(t)\sigma_+^{(j)} -\frac{1}{2}\{\sigma_+^{(j)}\sigma_-^{(i)},\rho(t)\}\right) \tag 1 $$
+with 
+$$ \begin{aligned}
+    A &= \begin{pmatrix}\alpha & \alpha+\frac{\kappa}{2}-i\frac{\gamma_2-\gamma_1}{4}\\ \alpha+\frac{\kappa}{2}-i\frac{\gamma_1-\gamma_2}{4} & \alpha+\kappa \end{pmatrix} \quad \text{and} \\
+    B &= \frac{1}{2}\begin{pmatrix}\gamma_1&\frac{\gamma_1+\gamma_2}{2}-i\kappa\\
+    \frac{\gamma_1+\gamma_2}{2}+i\,\kappa&\gamma_2 \end{pmatrix} ,
+\end{aligned} $$
+where $\alpha$ and $\kappa$ are real numbers and $\gamma_{1,2} \geq 0$.
+
+TODO: explain
+
+The matrix is self adjoint, and therefore it can be diagonalised:
+$$ \operatorname{diag}{B} = U^\dagger\, B\, U = \begin{pmatrix} \lambda_{1}&0\\0&\lambda_{2} \end{pmatrix} $$
+with $\lambda_{i} = \frac{\gamma_1+\gamma_2}{4}+(-1)^{i}\sqrt{\frac{\gamma_1^2+\gamma_2^2+2\,\kappa^2}{8}}$.
+We the define the Lindblad operators
+$$ L_j = \sum_{i=1}^2 \sigma_-^{(i)} U_{ij} $$
+such that equation (1) becomes
+$$ \frac{d}{dt}\rho(t) = -i \sum_{i,j=1}^2 A_{i,j} [\sigma_+^{(j)}\sigma_-^{(i)},\rho(t)] + \sum_{i}^2\lambda_i\left( L_i\rho(t)L_i^\dagger -\frac{1}{2}\{L_i^\dagger L_i,\rho(t)\}\right) . \tag 2 $$
+Now it is clear that when $\kappa>0$, $\lambda_1<0$ and thus Eq. (2) is not in strict Lindblad form.
+
+
+##### Setup system
+
+We choose the model parameters $\alpha$, $\kappa$ and $\gamma_{1,2}$, and calculate the rates $\lambda_1$ and $\lambda_2$.
+The rates are constant but must be provided to the solver as callable functions of time.
+
+```python
+gam1 = 1
+gam2 = 2
+alpha = 3
+kappa = 2
+
+lamb1 = (gam1 + gam2) / 4 - ((gam1**2 + gam2**2 + 2 * kappa) / 8) ** (1 / 2)
+lamb2 = (gam1 + gam2) / 4 + ((gam1**2 + gam2**2 + 2 * kappa) / 8) ** (1 / 2)
+
+
+def lam1(t):
+    return lamb1
```

Comment:
In the latest version of the branch, you can just supply the constant value instead of the function that returns a constant and the solver will convert it to the new `ConstantCoefficient`.